### PR TITLE
API provides list of pending tasks

### PR DESCRIPTION
**Pull request by Carltesio**
# [PT Story](https://www.pivotaltracker.com/story/show/172309398)
```
As a website visitor
in order to see the current pending tasks
I would like the backend to provide the list of tasks to the frontend
```
## Changes proposed in this pull request:
* Restricts to show only the confirmed tasks in the index method
* Updates route and controller to show a list of tasks
* Adds spec file to test confirmed tasks
## What I have learned working on this feature:
* Restricting content in the index action
* Sometimes you cannot finish the feature/task exactly as you want but it is important to find a good compromise to good working pace and quality.

### CODE REVIEW

File: spec/requests/api/v1/user_can_see_list_of_tasks_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    before do
+      get "/api/v1/tasks"
+    end
+
+    it "returns a 200 response status" do
+      expect(response).to have_http_status 200
+    end
+
+    it "returns correct number of tasks" do
+      expect(response_json.count).to eq 2
```
Comment (blake-futchi) [inline spec/requests/api/v1/user_can_see_list_of_tasks_spec.rb:24]: how is there 2 tasks from this?

File: app/serializers/task_serializer.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class TaskSerializer < ActiveModel::Serializer
-  attributes :id, :products, :total, :order_total
+  attributes :id, :products, :total
```
Comment (blake-futchi) [inline app/serializers/task_serializer.rb:2]: good, total and order total was confusing

File: app/models/task.rb [hunk 3]
```diff
@@ -1,7 +1,3 @@
 class Task < ApplicationRecord
   has_many :task_items
-
```
Comment (blake-futchi) [inline app/models/task.rb:3]: where are we creating the total order price now?

**Review by blake-futchi — COMMENTED** [review]
take a look at the notes i left

File: spec/requests/api/v1/user_can_see_list_of_tasks_spec.rb (hunk 1 above)
Comment (pierre-1) [reply spec/requests/api/v1/user_can_see_list_of_tasks_spec.rb:24]: The test expects 2 tasks to be created based on the two tasks created by the factorybot. One task is confirmed, the other is not.

**Review by blake-futchi — APPROVED** [review]
Great!

**Comment by blake-futchi** [comment]
boom

**Pull request merged**